src/Sulu/Bundle/WebsiteBundle/DataCollector/SuluCollector.php

```diff
@@ -76,39 +81,52 @@ public function collect(Request $request, Response $response, ?\Throwable $excep
-                /** @var StructureInterface $structureObject */
-                $structureObject = $params['structure'];
-
-                $structure = [
-                    'id' => $structureObject->getUuid(),
-                    'objectClass' => $structureObject::class,
-                    'path' => $structureObject->getPath(),
-                    'nodeType' => $structureObject->getNodeType(),
-                    'internal' => $structureObject->getInternal(),
-                    'nodeState' => $structureObject->getNodeState(),
-                    'published' => $structureObject->getPublished(),
-                    'publishedState' => $structureObject->getPublishedState(),
-                    'navContexts' => $structureObject->getNavContexts(),
-                    'shadowLocales' => $structureObject->getShadowLocales(),
-                    'contentLocales' => $structureObject->getContentLocales(),
-                    'shadowOn' => $structureObject->getIsShadow(),
-                    'shadowBaseLanguage' => $structureObject->getShadowBaseLanguage(),
-                    'template' => $structureObject->getKey(),
-                    'originTemplate' => $structureObject->getOriginTemplate(),
-                    'hasSub' => $structureObject->getHasChildren(),
-                    'creator' => $structureObject->getCreator(),
-                    'changer' => $structureObject->getChanger(),
-                    'created' => $structureObject->getCreated(),
-                    'changed' => $structureObject->getChanged(),
-                ];
+        if ($request->attributes->has('object')) {
+            $object = $request->attributes->get('object');
+
+            $structure = [];
+            if ($object instanceof DimensionContentInterface) {
+                $parent = $object->getResource();
+
+                if (\method_exists($parent, 'getUuid')) {
+                    $structure['id'] = $parent->getUuid();
+                }
+                $structure['class'] = $parent::class;
+                $structure['dimensionClass'] = $object::class;
+                $structure['nodeState'] = $object->getStage();
+                $structure['locale'] = $object->getLocale();
+                $structure['ghostLocale'] = $object->getGhostLocale();
+
+                if ($parent instanceof AuditableInterface) {
+                    $structure['created'] = $this->renderUserAndTimeStamp($parent->getCreator(), $parent->getCreated());
+                    $structure['changed'] = $this->renderUserAndTimeStamp($parent->getChanger(), $parent->getChanged());
+                }
+            }
+
+            if ($object instanceof TemplateInterface) {
+                $structure['template'] = $object->getTemplateKey();
+            }
+
+            if ($object instanceof WorkflowInterface) {
+                $structure['published'] = $object->getWorkflowPublished();
+            }
+
+            if (\method_exists($object, 'getNavigationContexts')) {
```

Comment:
For more specific things like `navigationContexts` or `hasChildren` we can check if it is an instance of `PageDimensionContentInterface` or `PageInterface`